src/calibrationlib/operator_feature_export.cpp

```diff
@@ -0,0 +1,107 @@
+  }
+}
+
+void OperatorFeatureExport::_export_typed_operator(std::shared_ptr<const AbstractOperator> op) const {
+  switch (op->type()) {
+    case OperatorType::TableScan:
+      _export_table_scan(op);
+      break;
+    default:
+      break;
+  }
+}
+
+// Export features of a table scan operator
+void OperatorFeatureExport::_export_table_scan(std::shared_ptr<const AbstractOperator> op) const {
+  DebugAssert(op->type() == OperatorType::TableScan, "Expected operator of type: TableScan but got another one");
+
+  auto csv_writer = _csv_writers.at(op->type());
+
+  const auto node = op->lqp_node;
+
+  for (const auto& el : node->node_expressions) {
+    visit_expression(el, [&](const auto& expression) {
+      if (expression->type == ExpressionType::LQPColumn) {
+        const auto column_expression = std::dynamic_pointer_cast<LQPColumnExpression>(expression);
+        const auto column_reference = column_expression->column_reference;
+        const auto original_node = column_reference.original_node();
+
+        if (original_node->type == LQPNodeType::StoredTable) {
+          // Get values for INPUT_ROWS_LEFT
+          if (op->input_left() != nullptr) {
+            csv_writer->set_value("INPUT_ROWS_LEFT", op->input_left()->performance_data().output_row_count);
+          } else {
+            csv_writer->set_value("INPUT_ROWS_LEFT", CSVWriter::NA);
+          }
+
+          // Get values for OUTPUT_ROWS & RUNTIME_NS
+          if (op->performance_data().has_output) {
+            csv_writer->set_value("OUTPUT_ROWS", op->performance_data().output_row_count);
+            // Get values for RUNTIME_NS
+            csv_writer->set_value("RUNTIME_NS", op->performance_data().walltime.count());
+          } else {
+            csv_writer->set_value("OUTPUT_ROWS", CSVWriter::NA);
+            csv_writer->set_value("RUNTIME_NS", CSVWriter::NA);
+          }
+
+          // Get values for SCAN_IMPLEMENTATION
+          if (original_node == node->left_input()) {
+            csv_writer->set_value("SCAN_TYPE", "COLUMN_SCAN");
+          } else {
+            csv_writer->set_value("SCAN_TYPE", "REFERENCE_SCAN");
+          }
+
+          // Get values for TABLE_NAME
+          const auto stored_table_node = std::dynamic_pointer_cast<const StoredTableNode>(original_node);
+          const auto& table_name = stored_table_node->table_name;
+
+          csv_writer->set_value("TABLE_NAME", table_name);
+
+          const auto original_column_id = column_reference.original_column_id();
+
+          // Get values for COLUMN_NAME
+          const auto table = Hyrise::get().storage_manager.get_table(table_name);
+          csv_writer->set_value("COLUMN_NAME", table->column_names()[original_column_id]);
+
+          // Get values for SCAN_IMPLEMENTATION
+          auto description = op->description();
+          std::smatch matches;
+
+          std::regex self_regex("Impl: ([A-Z]\\w+)", std::regex_constants::ECMAScript | std::regex_constants::icase);
```

Comment:
@Bouncner do you know a better way for this? 

Also, do we need ti define that again for every TableScan? It should stay the same, shouldn't it?

Reply:
Why is "ECMAScript" being used here? I'd be fine with a substring taking everything after `Impl: `.

Reply:
Definitely something to test for or Assert that the previous entry includes "Impl:".

Reply:
I added an Assert before I set the SCAN_IMPLEMENTATION value.